backend/controller/controller.go

```diff
@@ -1410,6 +1413,11 @@ func (s *Service) executeAsyncCalls(ctx context.Context) (time.Duration, error)
 	}
 	err = s.dal.CompleteAsyncCall(ctx, call, callResult, func(tx *dal.Tx) error {
 		if failed && call.RemainingAttempts > 0 {
+			// If the call failed and there are attempts remaining, then
+			// CompleteAsyncCall would enqueue another call, so increment
+			// queue depth accordingly.
+			queueDepth++
```

Comment:
This function locks (`s.asyncCallsLock.Lock()`) as it starts running, so I _think_ we're ok... I don't think there's any other code processing async calls. At least, at a glance, I couldn't find any, and there's a comment at the top of this function that says `There are multiple entry points into this function, but we want the controller to handle async calls one at a time.` Thanks for catching that, though, I forgot to double check this when I first wrote the PR.

Reply:
Ah, just wasn't sure how that lock works with this func as a parameter to `CompleteAsyncCall`

Reply:
Ah gotcha! So when you pass closures around in Go that reference local variables, the closure captures the enclosing scope. So since `CompleteAsyncCall` doesn't start a separate goroutine to run the closure in, we're all good :)

(as I say things like this I always panic a bit that the behavior's changed from the version of Go I learned this on to `1.22`)